hkmc2/shared/src/main/scala/hkmc2/codegen/Block.scala

```diff
@@ -552,7 +554,7 @@ type Local = Symbol
  * regardless of whether the check for effect is inserted or not.
  * Note that the check for effect is inserted during HandlerLowering and setting this to true
  * after handler is lowered does not have any effect on the code generation. */
-case class Call(fun: Path, args: Ls[Arg])(val isMlsFun: Bool, val mayRaiseEffects: Bool) extends Result
+case class Call(fun: Path, args: Ls[Arg])(val isMlsFun: Bool, val mayRaiseEffects: Bool, val isTailCall: Bool) extends Result
```

Comment:
I would name this
```suggestion
case class Call(fun: Path, args: Ls[Arg])(val isMlsFun: Bool, val mayRaiseEffects: Bool, val explicitTailCall: Bool) extends Result
```
as the call might not be in actual tail position (then, we'll either emit an error or transform the code to make it tail).